components/video-recorder/src/HeadlessVideoRecorder.tsx

```diff
@@ -0,0 +1,636 @@
+import React, {
+	useRef,
+	useState,
+	CSSProperties,
+	useCallback,
+	useEffect,
+	useImperativeHandle,
+} from "react";
+import { twMerge } from "tailwind-merge";
+import { useTimer } from "@locoworks/reusejs-toolkit-react-hooks";
+export interface HeadlessVideoRecorderRef {
+	recording: string;
+	handleDownload: () => void;
+	showPreview: () => void;
+}
+interface HeadlessVideoRecorderProps {
+	autoStop?: boolean;
+	autoPreview?: boolean;
+	timeInMs?: number;
+	handleStateChange?: (state: VideoState) => void;
+	customhandleDownload?: (file: any) => void;
+	customCountDown?: (count: number) => React.ReactNode;
+	customRetakeButton?: (onRetake: any) => React.ReactNode;
+	customStopRecording?: (onStop: any) => React.ReactNode;
+	customStartRecording?: (onStart: any) => React.ReactNode;
+	customDownloadButton?: (onDownload: any) => React.ReactNode;
+	mediaConstraints?: MediaStreamConstraints;
+	recorderContainerClasses?: string | CSSProperties;
+	loader?: React.ReactNode;
+	videoRecorderClasses?: string | CSSProperties;
+	playbackVideoClasses?: string | CSSProperties;
+	footerContainerClasses?: string | CSSProperties;
+	startRecordingbuttonClasses?: string | CSSProperties;
+	recordingFooterClasses?: string | CSSProperties;
+	previewFooterClasses?: string | CSSProperties;
+	recordedFooterClasses?: string | CSSProperties;
+	retakeButtonClasses?: string | CSSProperties;
+	downloadButtonClasses?: string | CSSProperties;
+	retakeLabel?: string;
+	downloadLabel?: string;
+	stopLabel?: string;
+	startLabel?: string;
+}
+
+export type VideoState = "inactive" | "preview" | "recording" | "recorded";
+
+type VideoMimeTypes = "video/mp4" | "video/webm";
```

Comment:
Is it possible to support other types of videos here? Will this support all the general devices?

Reply:
Handling video MIME types for MP4 and WebM covers a vast majority including desktops, laptops, and many mobile devices 
MP4 is supported across major browsers and devices, including desktops, laptops, iOS devices (iPhone, iPad), and most Android devices; and
WebM is Well-supported on modern browsers, including Chrome, Firefox, and Opera. On mobile, it's well-supported on Android devices, but iOS devices do not natively support WebM.